### PR TITLE
Remove collect

### PR DESCRIPTION
**Pull request by sklan**
Solves #657
```julia 
julia> xs = param([1, 2, 3])
Tracked 3-element Array{Float64,1}:
 1.0
 2.0
 3.0

julia> collect(xs)
3-element Array{Flux.Tracker.TrackedReal{Float64},1}:
 1.0
 2.0
 3.0
```

### CODE REVIEW

**Comment by MikeInnes** [comment]
Can you add a definition for `collect` that just returns `x`? Keeping this as a TrackedArray is more efficient.

**Comment by Sklan** [comment]
Yeah, sure. I'll do that.

**Comment by Sklan** [comment]
@MikeInnes It still keeps calling the ```Base.collect```
```julia
julia> xs = param([1, 2, 3])
Tracked 3-element Array{Float64,1}:
 1.0
 2.0
 3.0

julia> collect(xs)
3-element Array{Flux.Tracker.TrackedReal{Float64},1}:
 1.0
 2.0
 3.0

julia> @which collect(xs)
collect(A::AbstractArray) in Base at array.jl:546
```
I am confused. Am I missing something?

**Comment by MikeInnes** [comment]
You need to define `Base.collect` (otherwise it creates a new function, `Flux.collect`).

File: src/tracker/lib/array.jl [hunk 1]
```diff
@@ -71,9 +71,9 @@ function update!(x::TrackedArray, Δ)
   return x
 end
 
-collect(x::TrackedArray) = x
-collect(x::TrackedVector) = x
-collect(x::TrackedMatrix) = x
+Base.collect(x::TrackedArray) = x
+Base.collect(x::TrackedVector) = x
```
Comment (MikeInnes) [inline src/tracker/lib/array.jl]: The `Vector` and `Matrix` definitions are redundant here, since they are covered by `TrackedArray`.

**Comment by MikeInnes** [comment]
Unfortunately this got caught when we split out Tracker.jl. You should be able to easily PR the same branch against that repo, though.

**Comment by sklan** [comment]
Okay, I'll add a PR in Tracker.jl.

**Pull request closed without merging**